plugins/catalyst/python/pyproject.toml

```diff
@@ -0,0 +1,298 @@
+readme = "../../../README.md"
+authors = [
+    { name = "Patrick Hopf", email = "patrick.hopf@tum.de" },
+    { name = "Yannick Stade", email = "yannick.stade@tum.de" },
+    { name = "Lukas Burgholzer", email = "burgholzer@me.com" }
+]
+keywords = ["MQT", "quantum-computing", "design-automation", "catalyst", "MLIR", "plugin"]
+license = "MIT"
+# we have to duplicate the license file here because scikit_build_core requires
+# "the pattern [specifying the license file] must match files within the project directory"
+license-files = ["LICENSE.md"]
+
+classifiers = [
+  "Development Status :: 3 - Alpha",
+  "Intended Audience :: Science/Research",
+  "Natural Language :: English",
+  "Operating System :: MacOS",
+  "Operating System :: POSIX :: Linux",
+  "Programming Language :: C++",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Scientific/Engineering :: Electronic Design Automation (EDA)",
+  "Typing :: Typed",
+]
+requires-python = ">=3.10"
+dynamic = ["version"]
+
+[project.scripts]
+mqt-catalyst-cli = "mqt.catalyst.__main__:main"
+
+[project.urls]
+Homepage = "https://github.com/munich-quantum-toolkit/core"
+Changelog = "https://github.com/munich-quantum-toolkit/core/blob/main/CHANGELOG.md"
+Documentation = "https://mqt.readthedocs.io/projects/core"
+Issues = "https://github.com/munich-quantum-toolkit/core/issues"
+Discussions = "https://github.com/munich-quantum-toolkit/core/discussions"
+PyPI = "https://pypi.org/project/mqt-core/"
+
+
+[tool.scikit-build]
+# Protect the configuration against future changes in scikit-build-core
+minimum-version = "build-system.requires"
+
+# Set the wheel install directory
+wheel.install-dir = "mqt/catalyst"
+
+# Explicitly set the package directory
+wheel.packages = ["src/mqt"]
+
+# Set required Ninja version
+ninja.version = ">=1.10"
+
+# Setuptools-style build caching in a local directory
+build-dir = "../../../build/mqt-catalyst-plugin/{wheel_tag}/{build_type}"
+
+# All components to install
+install.components = [
+  "mqt-catalyst_Plugin",
+]
+
+metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
+sdist.include = ["src/mqt/catalyst/_version.py"]
+sdist.exclude = [
+  "**/docs",
+  "**/test",
+]
+
+[tool.check-sdist]
+sdist-only = ["src/mqt/catalyst/_version.py"]
+git-only = [
+  "docs/*",
+  "test/*",
+]
+
+[tool.scikit-build.cmake]
+source-dir = "../../.."
+
+[tool.scikit-build.cmake.define]
+BUILD_MQT_CORE_TESTS = "OFF"
+BUILD_MQT_CORE_SHARED_LIBS = "ON"
+BUILD_MQT_CORE_MLIR = "ON"
+BUILD_MQT_CORE_CATALYST_PLUGIN = "ON"
+# todo: How can that be done such that it works locally and in CI?
+MLIR_DIR = "/opt/homebrew/opt/llvm@19/lib/cmake/mlir"
+LLVM_EXTERNAL_LIT = "/opt/homebrew/opt/lit/bin/lit"
```

Comment:
This should also not be necessary here in this explicit fashion.
For the plugin build, we do not want to have some MLIR tests enabled anyway, so we should not need `lit` at all.